confgenerator/testdata/valid/linux/metrics-receiver_redis_missing_address/golden_otel.conf

```diff
@@ -426,6 +426,8 @@ receivers:
     collection_interval: 30s
     endpoint: localhost:6379
     password: pwd
+    tls:
+      insecure: true
```

Comment:
What integration test should be covering this change? I don't see any changes to integration tests in this PR.

Reply:
Good question.

The current integration test setup only test one case for each app on all representative distros. Might need some work there to support testing multiple scenarios for one application.

A workaround is to do something like https://github.com/GoogleCloudPlatform/ops-agent/tree/master/integration_test/third_party_apps_data/applications/shared_tls

To balance the test flakiness concerns and code coverage need, it's probably ok for the shared TLS util to trigger tests on one distro, assuming the underlying logic is identical across distros.

Does the upstream Open Telemetry collector have any TLS related tests?

Reply:
Doesn't look like it.

Reply:
Would you like the shared_TLS in this PR or is that a seperate PR to be made?

Reply:
Either way is OK. If this PR has been tested manually, we can get it in first then add integration tests in a follow up one.

Reply:
verified that it is working through manual testing again with the updates. 
![Screen Shot 2022-02-07 at 8 52 11 PM](https://user-images.githubusercontent.com/14144136/152902946-6325cbbe-0acf-4397-b404-c72b9c0a2b57.png)

Reply:
Thanks!